circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/Replica.java

```diff
@@ -203,7 +242,13 @@ public void updateMetadata(
       if (!statisticsToSet.isEmpty()) {
         LOG.info("Setting column statistics for {} partitions.", statisticsToSet.size());
         try {
-          client.setPartitionColumnStatistics(new SetPartitionsStatsRequest(statisticsToSet));
+          int counter = 0;
+          for (List<ColumnStatistics> sublist : Lists.partition(statisticsToSet, partitionBatchSize)) {
+            int start = counter * partitionBatchSize;
+            LOG.info("Setting column statistics for partitions {} through {}", start, start + sublist.size() - 1);
+            client.setPartitionColumnStatistics(new SetPartitionsStatsRequest(statisticsToSet));
```

Comment:
here too. `sublist` instead of `statisticsToSet` ?

Reply:
yep, thanks